File: answers/elixir/lib/aoc_helpers.ex
defmodule AocHelpers do
  @moduledoc """
  Helpers for advent of code problems solved with elixir.
  """

  @doc """
  Writes answers to a file named `answers.txt` in the current working directory.

  ## Examples

      iex> AocHelpers.write_answers!("answer 1", "answer 2")
      :ok

  """
  def write_answers!(answer1, answer2) do
    File.write!("answers.txt", format_answers(answer1, answer2))
    :ok
  end

  @doc """
  Reads lines from a file named `input.txt` in the current working directory.
  """
  def read_input!() do
    File.stream!("input.txt") |> Stream.map(&String.trim_trailing/1)
  end

  @doc """
  Formats the answers to be displayed in a readable way.
  """
  def format_answers(answer1, answer2) do
    ["Answer 1:\n", answer1, "\n\nAnswer 2:\n", answer2, "\n"]
  end
end

File: answers/go/aoc_helpers.go
package write_aoc_answers

import (
	"fmt"
	"os"
	"strings"
)

func WriteAnswers(first, second string) error {
	output := formatAnswers(first, second)
	err := os.WriteFile("answers.txt", []byte(output), 0666)
	if err != nil {
		return err
	}

	return nil
}

func ReadInput() ([]string, error) {
	output, err := os.ReadFile("input.txt")
	if err != nil {
		return nil, err
	}
	return strings.Split(string(output), "\n"), nil
}

func formatAnswers(first, second string) string {
	return fmt.Sprintf("Answer 1:\n%s\n\nAnswer 2:\n%s\n", first, second)
}
